File: tools/improve/tests/original/maudec/py/collatz.py


def rlapp_Nat(term: int) -> int | None:
	match term:
		case N if N % 2 == 0:
			return N // 2

		case N if N % 2 == 1:
			return 1 + N * 3

		case _:
			return None



def arlapp_Nat(term: int) -> int | None:
	return rlapp_Nat(term)

File: tools/improve/tests/original/maudec/rust/collatz.rs


fn rlapp_Nat(term: i32) -> Option<i32> {
	match term {
		n if n % 2 == 0 => Some(n / 2),
		n if n % 2 == 1 => Some(1 + n * 3),
		_ => None,
	}
}

fn arlapp_Nat(term: i32) -> Option<i32> {
	rlapp_Nat(term)
}

